examples/08_join_aggregation.py

```diff
@@ -1,302 +1,548 @@
-# logs.
+# We give it our trained model, and fit it on the validation dataset instead of the
+# training dataset to avoid overfitting. Notice that the scoring method is the utility
+# function, to which we pass the amount in ``fit``
+from sklearn.model_selection import TunedThresholdClassifierCV
+
+low_effort_tuned = TunedThresholdClassifierCV(
+    low_effort, cv="prefit", scoring=gain_score, refit=False
+).fit(X_val, y_val, amount=X_val["total_price"])
+
+results["Low effort"] = get_results(
+    low_effort,
+    X_test,
+    y_test,
+    threshold=low_effort_tuned.best_threshold_,
+    amount=X_test["total_price"],
+    time_to_fit=time_to_fit,
+)
+
+# %%
+# We define some plotting functions to display our results.
+import seaborn as sns
+from matplotlib import pyplot as plt
+from sklearn.calibration import CalibrationDisplay
+
+
+def plot_gain_tradeoff(results):
+    """Scatter plot of the score gain (y) vs the fit time (x) for each model."""
+
+    rows = []
+    for estimator_name, result in results.items():
+        result["estimator_name"] = estimator_name
+        rows.append(result)
+    df = pd.DataFrame(rows)
+
+    names = df["estimator_name"].values
+    palette = dict(zip(names, sns.color_palette("colorblind", n_colors=len(names))))
+
+    fig, ax = plt.subplots(figsize=(5, 4), dpi=100)
+    sns.scatterplot(
+        df,
+        x="time_to_fit",
+        y="gain_score",
+        hue="estimator_name",
+        style="estimator_name",
+        ax=ax,
+        palette=palette,
+        s=200,
+    )
+    ax.grid()
+
+    ticks = df["time_to_fit"].round(3).tolist()
+    labels = [f"{tick}s" for tick in ticks]
+    ax.set_xticks(ticks, labels)
+
+    ticks = df["gain_score"].round().tolist()
+    ticks.insert(1, 650_000)
+    labels = [f"{tick:,} €" for tick in ticks]
+
+    ax.set_yticks(ticks, labels)
+    ax.set_ylabel("Gain score")
+    ax.set_xlabel("Time to fit")
+    ax.set_title("Gain score vs Time to fit")
+    plt.tight_layout()
+
+
+def plot_calibration_curve(results):
+    """Plot a calibration curve and the log-loss."""
+
+    estimator_names = list(results)
+    palette = dict(
+        zip(
+            estimator_names,
+            sns.color_palette("colorblind", n_colors=len(estimator_names)),
+        )
+    )
+    fig, ax = plt.subplots(figsize=(6, 4), dpi=100)
+    for name, result in results.items():
+        log_loss = str(round(result["log_loss"], 4))
+        label = f"{name}, {'log_loss: ' + log_loss}"
+        CalibrationDisplay.from_predictions(
+            y_true=result["y_test"],
+            y_prob=result["y_proba"],
+            strategy="quantile",
+            label=label,
+            ax=ax,
+            color=palette[name],
+            n_bins=15,
+        )
+    ax.set_xlim([-0.001, 0.13])
+    ax.set_ylim([-0.001, 0.13])
+    ax.set_title("Calibration curve")
+
+
+# %%
+# We see below that the low effort classifier significantly improves our gains compared
```

Comment:
I would maybe remind the reader that there aren't that many frauds and we show the total gain so even if we completely ignore frauds we still have a high positive gain. otherwise we may think that there is a difference but in proportion it is not huge